lib/stat_tracker.rb

```diff
@@ -119,13 +118,18 @@ def games_won_per_team_for(season_id)
     end
   end
 
-  def shots_to_goals_ratio_per_team
-    total_shots_per_team.merge(total_goals_per_team){|team_id, shots, goals| (shots.to_f / goals).round(2)}
+  def shots_to_goals_ratio_per_team(season_id)
+    total_shots_per_team_per_season(season_id).merge(total_goals_per_team_for_season(season_id)){|team_id, shots, goals| (shots.to_f / goals).round(2)}
   end
-
-  def total_games_per_team_for(season_id)
+  def total_goals_per_team_for_season(season_id)
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += game_team.goals if game_team.season == season_id
+      result
+    end
+  end
+  def total_shots_per_team_per_season(season_id)
     @game_teams.reduce(Hash.new(0)) do |result, game_team|
-      result[game_team.team_id] += 1 if game_team.season == season_id
+      result[game_team.team_id] += game_team.shots if game_team.season == season_id
       result
     end
   end
```

Comment:
same here! for_season instead of per season

Reply:
All fixed!